### PR TITLE
Fix github artifact upload using explicit jar names

### PR DESCRIPTION
**Pull request by randomanderson**
# What Does This Do

Fixes the github artifact upload by using explicit file names

# Motivation
The previous way of using `*.jar` is broken because there are `-source.jar` and `-javadoc.jar` when using artifacts from the `deploy_to_sonotype` job. We know the exact name of the files because it is only run when there is an explicit `CI_COMMIT_TAG` set.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -236,11 +236,9 @@ deploy_artifacts_to_github:
   rules:
     - if: '$POPULATE_CACHE'
       when: never
-    - if: '$CI_COMMIT_TAG =~ /^v.*/'
+    - if: '$CI_COMMIT_TAG =~ /^v[0-9]+\.[0-9]+\.[0-9]+$/'
```
Comment (randomanderson) [inline .gitlab-ci.yml:239]: This needs to exactly match the `deploy_to_sonotype` condition above

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -236,11 +236,9 @@ deploy_artifacts_to_github:
...
       when: on_success
-    - when: manual
```
Comment (randomanderson) [inline .gitlab-ci.yml:241]: `manual` doesn't make sense because we use `CI_COMMIT_TAG` in the commands

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -250,10 +248,11 @@ deploy_artifacts_to_github:
     - aws ssm get-parameter --region us-east-1 --name ci.dd-trace-java.gh_release_token --with-decryption --query "Parameter.Value" --out text > github-token.txt
     - gh auth login --with-token < github-token.txt
     - gh auth status  # Maybe helpful to have this output in logs?
-    - cp workspace/dd-java-agent/build/libs/dd-java-agent-*.jar workspace/dd-java-agent/build/libs/dd-java-agent.jar # we upload two filenames
-    - gh release upload --clobber --repo DataDog/dd-trace-java $CI_COMMIT_TAG workspace/dd-java-agent/build/libs/*.jar
-    - gh release upload --clobber --repo DataDog/dd-trace-java $CI_COMMIT_TAG workspace/dd-trace-api/build/libs/*.jar
-    - gh release upload --clobber --repo DataDog/dd-trace-java $CI_COMMIT_TAG workspace/dd-trace-ot/build/libs/*.jar
+    - cp workspace/dd-java-agent/build/libs/dd-java-agent-${CI_COMMIT_TAG}.jar workspace/dd-java-agent/build/libs/dd-java-agent.jar # we upload two filenames
```
Comment (PerfectSlayer) [inline .gitlab-ci.yml:251]: Isn't CI commit tag `v1.37.0` where artifact name are `dd-java-agent-1.37.0`?
There is no `v` prefix in front of the version number.

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -250,10 +248,11 @@ deploy_artifacts_to_github:
...
+    - gh release upload --clobber --repo DataDog/dd-trace-java $CI_COMMIT_TAG workspace/dd-java-agent/build/libs/dd-java-agent.jar
+    - gh release upload --clobber --repo DataDog/dd-trace-java $CI_COMMIT_TAG workspace/dd-java-agent/build/libs/dd-java-agent-${CI_COMMIT_TAG}.jar
+    - gh release upload --clobber --repo DataDog/dd-trace-java $CI_COMMIT_TAG workspace/dd-trace-api/build/libs/dd-trace-api-${CI_COMMIT_TAG}.jar
+    - gh release upload --clobber --repo DataDog/dd-trace-java $CI_COMMIT_TAG workspace/dd-trace-ot/build/libs/dd-trace-ot-${CI_COMMIT_TAG}.jar
```
Comment (PerfectSlayer) [inline .gitlab-ci.yml:255]: `gh release upload` is able to take multiple files from the same command.
Is it better to make a single call or multiple ones? (I honestly don't know)

**Review by PerfectSlayer — COMMENTED** [review]
There might be an issue with the version name.

File: .gitlab-ci.yml (hunk 3 above)
Comment (randomanderson) [reply .gitlab-ci.yml:251]: Good catch

File: .gitlab-ci.yml (hunk 4 above)
Comment (randomanderson) [reply .gitlab-ci.yml]: I think it's clearer to have multiple lines. I don't think there's a difference functionally

File: .gitlab-ci.yml (hunk 3 above)
Comment (randomanderson) [reply .gitlab-ci.yml]: Fixed

**Review by PerfectSlayer — APPROVED** [review]
Looks good now!
Don't forget to squash on merge 👍

**Pull request merged**